staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -404,7 +429,7 @@ func validateCustomResourceDefinitionSpec(ctx context.Context, spec *apiextensio
 	}
 
 	allErrs = append(allErrs, ValidateCustomResourceDefinitionNames(&spec.Names, fldPath.Child("names"))...)
-	allErrs = append(allErrs, validateCustomResourceDefinitionValidation(ctx, spec.Validation, hasAnyStatusEnabled(spec), opts, fldPath.Child("validation"))...)
+	allErrs = append(allErrs, validateCustomResourceDefinitionValidation(ctx, spec.Validation, hasAnyStatusEnabled(spec), false, opts, fldPath.Child("validation"))...)
```

Comment:
alternatively, we're already plumbing `opts` all the way down ... you could put the `suppressPerExpressionCost` option in that struct instead of plumbing a separate boolean everywhere, and pass a copy of the opts struct with that populated down into `validateCustomResourceDefinitionValidation`

```suggestion
	allErrs = append(allErrs, validateCustomResourceDefinitionValidation(ctx, spec.Validation, hasAnyStatusEnabled(spec), suppressExpressionCostForUnchangedSchema(opts, spec.Version), fldPath.Child("validation"))...)
```



```go
// returns a copy of opts with suppressPerExpressionCost set to true if the specified version's schema is unchanged
func suppressExpressionCostForUnchangedSchema(opts validationOptions, version string) validationOptions {
  if opts.versionsWithUnchangedSchemas.Has(version) {
    opts.suppressPerExpressionCost = true
  }
  return opts
}
```

Reply:
Thank you for the advice! I've dropped the extra `suppressExpressionCostForUnchangedSchema` param.